Sources/NIOHTTPCompression/HTTPResponseCompressor.swift

```diff
@@ -84,11 +103,23 @@ public final class HTTPResponseCompressor: ChannelDuplexHandler, RemovableChanne
     private var pendingWritePromise: EventLoopPromise<Void>!
 
     private let initialByteBufferCapacity: Int
+    private let responseCompressionPredicate: ResponseCompressionPredicate?
 
-    /// Initialise a ``HTTPResponseCompressor``
+    /// Initialize a ``HTTPResponseCompressor``.
+    /// - Parameter initialByteBufferCapacity: Initial size of buffer to allocate when hander is first added.
+    @_disfavoredOverload
+    @available(*, deprecated, message: "Deprecated in favor of HTTPResponseCompressor(initialByteBufferCapacity:, responseCompressionPredicate:)")
+    public convenience init(initialByteBufferCapacity: Int = 1024) {
+        // TODO: This version is kept around for backwards compatibility and should be merged with the signature below in the next major version.
```

Comment:
Not sure what the best practice is here for deprecating overloads that can break backwards compatibility. I remember this was a thing early on in Swift 5, but not sure if its still relevant?

Reply:
Let's not deprecate this: we can just have two methods and remove the default argument for the new one.

Reply:
Should I keep the todo? Is there another mechanism to not forget these on the next major version bump?

Reply:
Please go ahead and file an issue and we can mark it for the next major version, but I'm happy to keep the TODO as well.

Reply:
Linked in the code as well: https://github.com/apple/swift-nio-extras/issues/226